### PR TITLE
List of current changes:

### PR DESCRIPTION
**Pull request by IgorKryviuk**
1.	Within the common-modules module:
        1)	change Cart class;
        2)	add History class;
        3)	delete ProductAmountItem class.
    2.	Within the RESTserver module:
        1)	add CartDAO;
        2)	add HistoryDAO class;
        3)	rename resources\application.properties file to database.properties file and make appropriate changes in .gitignore and README.md files;
        4)	add maintaining of a pooled data sorce (Apache Commons DBCP BasicDataSource) instead of simple DriverManagerDataSource.

### CODE REVIEW

File: RESTserver/src/main/java/com/softserve/if072/restservice/dao/mybatisdao/CartDAO.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
+package com.softserve.if072.restservice.dao.mybatisdao;
+
+import com.softserve.if072.common.model.Cart;
+import com.softserve.if072.common.model.Product;
+import com.softserve.if072.common.model.Store;
+import com.softserve.if072.common.model.User;
+import org.apache.ibatis.annotations.*;
```
Comment (kucherif) [inline RESTserver/src/main/java/com/softserve/if072/restservice/dao/mybatisdao/CartDAO.java]: please specify imports

File: RESTserver/src/main/java/com/softserve/if072/restservice/dao/mybatisdao/HistoryDAO.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
+package com.softserve.if072.restservice.dao.mybatisdao;
+
+import com.softserve.if072.common.model.History;
+import com.softserve.if072.common.model.Product;
+import com.softserve.if072.common.model.User;
+import org.apache.ibatis.annotations.*;
```
Comment (kucherif) [inline RESTserver/src/main/java/com/softserve/if072/restservice/dao/mybatisdao/HistoryDAO.java]: and here

Comment (IgorKryviuk) [inline RESTserver/src/main/java/com/softserve/if072/restservice/dao/mybatisdao/HistoryDAO.java]: have done

File: RESTserver/src/main/java/com/softserve/if072/restservice/dao/mybatisdao/CartDAO.java (hunk 1 above)
Comment (IgorKryviuk) [inline RESTserver/src/main/java/com/softserve/if072/restservice/dao/mybatisdao/CartDAO.java]: have done

**Pull request merged**